src/mumble/AudioInput.cpp

```diff
@@ -972,20 +972,26 @@ void AudioInput::encodeAudioFrame(AudioChunk chunk) {
 		iHoldFrames = 0;
 	}
 
+	// If Global::get().iPushToTalk > 0 that means that we are currently in some sort of PTT action. For
+	// instance this could mean we're currently whispering
+	bool isPTT = Global::get().iPushToTalk > 0;
+
 	if (Global::get().s.atTransmit == Settings::Continuous
 		|| API::PluginData::get().overwriteMicrophoneActivation.load()) {
 		// Continuous transmission is enabled
 		bIsSpeech = true;
 	} else if (Global::get().s.atTransmit == Settings::PushToTalk) {
 		// PTT is enabled, so check if it is currently active
-		bIsSpeech = Global::get().s.uiDoublePush
-					&& ((Global::get().uiDoublePush < Global::get().s.uiDoublePush)
-						|| (Global::get().tDoublePush.elapsed() < Global::get().s.uiDoublePush));
+		bool doublePush = Global::get().s.uiDoublePush
+						  && ((Global::get().uiDoublePush < Global::get().s.uiDoublePush)
+							  || (Global::get().tDoublePush.elapsed() < Global::get().s.uiDoublePush));
+
+		// With double push enabled, we might be in a PTT state without pressing any PTT key
+		bIsSpeech = doublePush;
+		isPTT     = isPTT || doublePush;
```

Comment:
1. Shouldn't `bIsSpeech` also depend on `isPTT`?
2. Given that we're in an explicit PTT-block, wouldn't it make more sense to set `bIsSpeech`accordingly and then at the end simply do a `isPTT = IsSpeech` as the only way to create speech input right now is by means of PTT. Or am I  missing something?

Reply:
The input method is very confusing (it was before i was touching it)

bIsSpeech depends on isPTT just further down. The reason for this is, as far as i can tell, that isPTT can occur when not using PTT as transmit by pressing the PTT key/ shout/ whisper etc.

We cant leave bIsSpeech out of the PTT block, because it is set by VAD before independently of the audio transmit method so it has to be explicitly set to false/the value of doublePush by the PTT block. This was the bug I encountered earlier where voice activation was still happening when selecting PTT.

The correct solution would be to refactor the entire method, which I would currently prefer not to do :D

Reply:
If I understand the code correctly here, we should be able to use
```suggestion
		isPTT     = isPTT || doublePush;
		bIsSpeech = isPTT;
```

The reason why I think this makes more sense semantically is that in PTT mode it seems logical to first check whether PTT is active and then `bIsSpeech` directly depends on that result.
This just seems to give the code local to this branch some self-containment as it does not depend on `bIsSpeech` also depending on `isPTT` further down.

But yeah - the audio processing code is pretty messy and unfortunately rewriting it will be a lot of effort. However, we are slowly working towards that goal: https://github.com/mumble-voip/libcrossaudio